fs/span-manager/span_manager.go

```diff
@@ -465,15 +461,22 @@ func (m *SpanManager) uncompressSpan(s *span, compressedBuf []byte) ([]byte, err
 }
 
 // addSpanToCache adds contents of the span to the cache.
-func (m *SpanManager) addSpanToCache(spanID compression.SpanID, contents []byte, opts ...cache.Option) {
-	if w, err := m.cache.Add(fmt.Sprintf("%d", spanID), opts...); err == nil {
-		if n, err := w.Write(contents); err != nil || n != len(contents) {
-			w.Abort()
-		} else {
-			w.Commit()
-		}
-		w.Close()
+// A non-nil error is returned if the data is not written to the cache.
+func (m *SpanManager) addSpanToCache(spanID compression.SpanID, contents []byte, opts ...cache.Option) error {
+	w, err := m.cache.Add(fmt.Sprintf("%d", spanID), opts...)
+	if err != nil {
+		return err
 	}
+	defer w.Close()
+
+	_, err = w.Write(contents)
```

Comment:
Should we be checking the number of bytes written here like in the previous version?

Reply:
From the [`io.Writer` interface doc](https://pkg.go.dev/io#Writer):

"It returns the number of bytes written from p (0 <= n <= len(p)) and any error encountered that caused the write to stop early. **Write must return a non-nil error if it returns n < len(p)**".